tools/cache.py

```diff
@@ -177,8 +177,6 @@ def get(shortname, creator, what=None, force=False, quiet=False, deferred=False)
     logger.info(message)
     utils.safe_ensure_dirs(cachename.parent)
     creator(str(cachename))
-    if not deferred:
```

Comment:
The removal of this assertion is also a consequence of localizing the change to __init__.py.

Reply:
This code is generic to all caching so it seems unfortunate not to be able to assert that the result of a build actually creates the file.   Can we revert this part?

Reply:
We can; I thought it was slightly ugly to thread `deferred` as an argument through all the port generators so I removed it, but since we put in the env var, we can just use that instead.

Reply:
But this still isn't great since we are saying that we don't care about the result of any cache access when `EMBUILDER_PORT_BUILD_DEFERRED` is set.

I guess its better than not checking at all though.

Are you sure we need this, given that `port.get` is not longer called from `add_cflags` at all?

Reply:
Yeah, this also affects non-port system libs builds (it's why this assertion was already conditional and not enabled with deferred builds before this PR). If we wanted something equivalent for deferred builds, we'd have to put code somewhere (here, presumably) to accumulate the list of deferred files we eventually want to see in the cache and verify that they all exist after the ninja invocation finishes. We could maybe introduce an API like `cache.verify_deferred` or something like that.